src/main/java/de/tum/in/www1/artemis/service/user/UserService.java

```diff
@@ -434,10 +443,19 @@ protected void deleteUser(User user) {
         // 8) Remove the user from its teams
         // 9) Delete the submissionVersion / remove the user from the submissionVersion
         // 10) Delete the tutor participation
+        // 11) All tutorial group registrations of the student
+        // 12) All tutorial groups taught by the user
 
         studentScoreRepository.deleteAllByUser(user);
         exerciseHintActivationRepository.deleteAllByUser(user);
 
+        tutorialGroupRegistrationRepository.deleteAllByStudent(user);
```

Comment:
Why do you do this manually here instead of using a cascade annotation?

Reply:
I tried it and it did not work with the cascade annotation alone.